### ISSUE TITLE
Could not initialize class org.apache.http.conn.ssl.SSLConnectionSocketFactory

### ISSUE THREAD

**Issue by dneves57** [apache/netbeans#7436]
### Apache NetBeans version

Apache NetBeans 22

### What happened

I have the following exception in Netbeans;
```

SEVERE [org.openide.util.Exceptions]
java.lang.ExceptionInInitializerError: Exception java.lang.ExceptionInInitializerError [in thread "RepositoryUpdater.worker"]
	at org.apache.http.conn.ssl.SSLConnectionSocketFactory.<clinit>(SSLConnectionSocketFactory.java:151)
	at org.eclipse.aether.transport.http.GlobalState.newConnectionManager(GlobalState.java:169)
	at org.eclipse.aether.transport.http.LocalState.<init>(LocalState.java:57)
	at org.eclipse.aether.transport.http.HttpTransporter.<init>(HttpTransporter.java:197)
	at org.eclipse.aether.transport.http.HttpTransporterFactory.newInstance(HttpTransporterFactory.java:95)
	at org.eclipse.aether.internal.impl.DefaultTransporterProvider.newTransporter(DefaultTransporterProvider.java:94)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.<init>(BasicRepositoryConnector.java:128)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory.newInstance(BasicRepositoryConnectorFactory.java:172)
	at org.eclipse.aether.internal.impl.DefaultRepositoryConnectorProvider.newRepositoryConnector(DefaultRepositoryConnectorProvider.java:122)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:536)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:449)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:261)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:243)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveArtifact(DefaultRepositorySystem.java:278)
	at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:197)
	at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolveAlways(DefaultArtifactResolver.java:152)
	at org.netbeans.modules.maven.embedder.MavenEmbedder.resolveArtifact(MavenEmbedder.java:424)
	at org.netbeans.modules.maven.j2ee.MavenJsfReferenceImplementationProvider.artifactPathFor(MavenJsfReferenceImplementationProvider.java:81)
	at org.netbeans.modules.web.jsf.editor.facelets.FaceletsLibrarySupport.parseLibraries(FaceletsLibrarySupport.java:361)
	at org.netbeans.modules.web.jsf.editor.facelets.FaceletsLibrarySupport._findLibraries(FaceletsLibrarySupport.java:325)
	at org.netbeans.modules.web.jsf.editor.facelets.FaceletsLibrarySupport.findLibraries(FaceletsLibrarySupport.java:270)
	at org.netbeans.modules.web.jsf.editor.facelets.FaceletsLibrarySupport.getNamespaceLibraryMapping(FaceletsLibrarySupport.java:173)
	at org.netbeans.modules.web.jsf.editor.JsfSupportImpl.getLibrary(JsfSupportImpl.java:224)
	at org.netbeans.modules.web.jsf.editor.JsfPageMetadataProvider.getMetadataMap(JsfPageMetadataProvider.java:74)
	at org.netbeans.modules.web.common.api.WebPageMetadata.getMetadata(WebPageMetadata.java:54)
	at org.netbeans.modules.web.common.api.WebPageMetadata.getMetadata(WebPageMetadata.java:113)
	at org.netbeans.modules.web.common.api.WebPageMetadata.getContentMimeType(WebPageMetadata.java:87)
	at org.netbeans.modules.html.editor.HtmlErrorFilter.isErrorCheckingEnabledForMimetype(HtmlErrorFilter.java:147)
	at org.netbeans.modules.html.editor.hints.HtmlHintsProvider.computeErrors(HtmlHintsProvider.java:228)
	at org.netbeans.modules.html.editor.HtmlErrorFilter.filter(HtmlErrorFilter.java:82)
	at org.netbeans.modules.csl.core.ErrorFilterQuery.getFilteredErrors(ErrorFilterQuery.java:57)
	at org.netbeans.modules.csl.core.TLIndexerFactory$TLIndexer.index(TLIndexerFactory.java:255)
	at org.netbeans.modules.parsing.spi.indexing.Indexable$MyAccessor$3.run(Indexable.java:225)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater.runIndexer(RepositoryUpdater.java:274)
	at org.netbeans.modules.parsing.spi.indexing.Indexable$MyAccessor.index(Indexable.java:223)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work$1T.run(RepositoryUpdater.java:3229)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work$1T.run(RepositoryUpdater.java:3249)
	at org.netbeans.modules.parsing.impl.TaskProcessor.callUserTask(TaskProcessor.java:586)
	at org.netbeans.modules.parsing.api.ParserManager$MultiUserTaskAction.run(ParserManager.java:235)
	at org.netbeans.modules.parsing.api.ParserManager$MultiUserTaskAction.run(ParserManager.java:205)
	at org.netbeans.modules.parsing.impl.TaskProcessor$2.call(TaskProcessor.java:181)
	at org.netbeans.modules.parsing.impl.TaskProcessor$2.call(TaskProcessor.java:178)
	at org.netbeans.modules.masterfs.filebasedfs.utils.FileChangedManager.priorityIO(FileChangedManager.java:153)
	at org.netbeans.modules.masterfs.providers.ProvidedExtensions.priorityIO(ProvidedExtensions.java:335)
	at org.netbeans.modules.parsing.nb.DataObjectEnvFactory.runPriorityIO(DataObjectEnvFactory.java:118)
	at org.netbeans.modules.parsing.impl.Utilities.runPriorityIO(Utilities.java:67)
	at org.netbeans.modules.parsing.impl.TaskProcessor.runUserTask(TaskProcessor.java:178)
	at org.netbeans.modules.parsing.api.ParserManager.parse(ParserManager.java:85)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work.indexEmbedding(RepositoryUpdater.java:3268)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work.doIndex(RepositoryUpdater.java:2861)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work.lambda$index$0(RepositoryUpdater.java:2626)
	at org.netbeans.modules.parsing.impl.indexing.errors.TaskCache.refreshTransaction(TaskCache.java:540)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work.index(RepositoryUpdater.java:2625)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$AbstractRootsWork.lambda$scanSource$3(RepositoryUpdater.java:5735)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater.lambda$runInContext$4(RepositoryUpdater.java:2119)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:288)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater.runInContext(RepositoryUpdater.java:2117)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater.runInContext(RepositoryUpdater.java:2098)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater.access$1400(RepositoryUpdater.java:135)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$AbstractRootsWork.scanSource(RepositoryUpdater.java:5770)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$AbstractRootsWork.scanSources(RepositoryUpdater.java:5443)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$RootsWork.getDone(RepositoryUpdater.java:5075)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work.doTheWork(RepositoryUpdater.java:3452)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Task._run(RepositoryUpdater.java:6197)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Task.access$3400(RepositoryUpdater.java:5855)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Task$2.lambda$call$0(RepositoryUpdater.java:6116)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:288)
	at org.netbeans.modules.parsing.impl.RunWhenScanFinishedSupport.performScan(RunWhenScanFinishedSupport.java:83)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Task$2.call(RepositoryUpdater.java:6116)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Task$2.call(RepositoryUpdater.java:6112)
	at org.netbeans.modules.masterfs.filebasedfs.utils.FileChangedManager.priorityIO(FileChangedManager.java:153)
	at org.netbeans.modules.masterfs.providers.ProvidedExtensions.priorityIO(ProvidedExtensions.java:335)
	at org.netbeans.modules.parsing.nb.DataObjectEnvFactory.runPriorityIO(DataObjectEnvFactory.java:118)
	at org.netbeans.modules.parsing.impl.Utilities.runPriorityIO(Utilities.java:67)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Task.run(RepositoryUpdater.java:6112)
Caused: java.lang.NoClassDefFoundError: Could not initialize class org.apache.http.conn.ssl.SSLConnectionSocketFactory
	at org.eclipse.aether.transport.http.GlobalState.newConnectionManager(GlobalState.java:169)
	at org.eclipse.aether.transport.http.LocalState.<init>(LocalState.java:57)
	at org.eclipse.aether.transport.http.HttpTransporter.<init>(HttpTransporter.java:197)
	at org.eclipse.aether.transport.http.HttpTransporterFactory.newInstance(HttpTransporterFactory.java:95)
	at org.eclipse.aether.internal.impl.DefaultTransporterProvider.newTransporter(DefaultTransporterProvider.java:94)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.<init>(BasicRepositoryConnector.java:128)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory.newInstance(BasicRepositoryConnectorFactory.java:172)
	at org.eclipse.aether.internal.impl.DefaultRepositoryConnectorProvider.newRepositoryConnector(DefaultRepositoryConnectorProvider.java:122)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:536)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:449)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:261)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:243)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveArtifact(DefaultRepositorySystem.java:278)
	at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:197)
	at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolveAlways(DefaultArtifactResolver.java:152)
	at org.netbeans.modules.maven.embedder.MavenEmbedder.resolveArtifact(MavenEmbedder.java:424)
	at org.netbeans.modules.maven.j2ee.MavenJsfReferenceImplementationProvider.artifactPathFor(MavenJsfReferenceImplementationProvider.java:81)
	at org.netbeans.modules.web.jsf.editor.facelets.FaceletsLibrarySupport.parseLibraries(FaceletsLibrarySupport.java:361)
	at org.netbeans.modules.web.jsf.editor.facelets.FaceletsLibrarySupport._findLibraries(FaceletsLibrarySupport.java:325)
	at org.netbeans.modules.web.jsf.editor.facelets.FaceletsLibrarySupport.findLibraries(FaceletsLibrarySupport.java:270)
	at org.netbeans.modules.web.jsf.editor.facelets.FaceletsLibrarySupport.getNamespaceLibraryMapping(FaceletsLibrarySupport.java:173)
	at org.netbeans.modules.web.jsf.editor.JsfSupportImpl.getLibrary(JsfSupportImpl.java:224)
	at org.netbeans.modules.web.jsf.editor.JsfPageMetadataProvider.getMetadataMap(JsfPageMetadataProvider.java:74)
	at org.netbeans.modules.web.common.api.WebPageMetadata.getMetadata(WebPageMetadata.java:54)
	at org.netbeans.modules.web.common.api.WebPageMetadata.getMetadata(WebPageMetadata.java:113)
	at org.netbeans.modules.web.common.api.WebPageMetadata.getContentMimeType(WebPageMetadata.java:87)
	at org.netbeans.modules.html.editor.HtmlErrorFilter.isErrorCheckingEnabledForMimetype(HtmlErrorFilter.java:147)
	at org.netbeans.modules.html.editor.HtmlErrorFilter.isErrorCheckingEnabled(HtmlErrorFilter.java:138)
	at org.netbeans.modules.html.editor.hints.HtmlHintsProvider.computeSuggestions(HtmlHintsProvider.java:146)
	at org.netbeans.modules.csl.hints.infrastructure.SuggestionsTask$1.run(SuggestionsTask.java:131)
	at org.netbeans.modules.parsing.impl.TaskProcessor.callUserTask(TaskProcessor.java:586)
	at org.netbeans.modules.parsing.api.ParserManager$UserTaskAction.run(ParserManager.java:197)
	at org.netbeans.modules.parsing.api.ParserManager$UserTaskAction.run(ParserManager.java:180)
	at org.netbeans.modules.parsing.impl.TaskProcessor$2.call(TaskProcessor.java:181)
	at org.netbeans.modules.parsing.impl.TaskProcessor$2.call(TaskProcessor.java:178)
	at org.netbeans.modules.masterfs.filebasedfs.utils.FileChangedManager.priorityIO(FileChangedManager.java:153)
	at org.netbeans.modules.masterfs.providers.ProvidedExtensions.priorityIO(ProvidedExtensions.java:335)
	at org.netbeans.modules.parsing.nb.DataObjectEnvFactory.runPriorityIO(DataObjectEnvFactory.java:118)
	at org.netbeans.modules.parsing.impl.Utilities.runPriorityIO(Utilities.java:67)
	at org.netbeans.modules.parsing.impl.TaskProcessor.runUserTask(TaskProcessor.java:178)
	at org.netbeans.modules.parsing.api.ParserManager.parse(ParserManager.java:83)
	at org.netbeans.modules.csl.hints.infrastructure.SuggestionsTask.run(SuggestionsTask.java:97)
	at org.netbeans.modules.csl.hints.infrastructure.SuggestionsTask.run(SuggestionsTask.java:54)
	at org.netbeans.modules.parsing.impl.TaskProcessor.callParserResultTask(TaskProcessor.java:561)
	at org.netbeans.modules.parsing.impl.TaskProcessor$RequestPerformer.run(TaskProcessor.java:786)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:288)
	at org.netbeans.modules.parsing.impl.TaskProcessor$RequestPerformer.execute(TaskProcessor.java:702)
[catch] at org.netbeans.modules.parsing.impl.TaskProcessor$CompilationJob.run(TaskProcessor.java:663)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1420)
	at org.netbeans.modules.openide.util.GlobalLookup.execute(GlobalLookup.java:45)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:287)
	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2035)

```

After compiling the project everything is fine, but the xhtml files cannot scan the libraries and load de components of primefaces, and this exception appears.

### Language / Project Type / NetBeans Component

Java Maven Web application project

### How to reproduce

Project with jdk22, primefaces and jakarta.

### Did this work correctly in an earlier version?

Apache NetBeans 21

### Operating System

Windows

### JDK

JDK 22

### Apache NetBeans packaging

Apache NetBeans provided installer

### Are you willing to submit a pull request?

No

**Comment by SimonePadovan** [comment]
I have same problem using NB22 on JDK 17 on Windows
When i enter the xhtml (jsf) page for editing it comes out this error and the editor is not able to load the facelet specification in order to autocmplete parameters of the tags or orher autocompletion features.

**Comment by matthiasblaesing** [comment]
This might be a duplicate of #7354. What is missing here are steps to reproduce. Without these, this issue won't be fixed. So please see what is necessary to reach the problem state starting from a clean NetBeans Installation.

**Comment by SimonePadovan** [comment]
Here are my steps: install Netbeans 22 on JDK 17, open project developed using NB < 22, enter editor of xhtml file. Nothing else. 
A collegue of mine has made the same steps but the problem doenst occour to him, tried to import his configuration but no success.
If i create a brand new maven web jsf project, the editor works but after restart of ntebeans it doesnt work again

**Comment by dneves57** [comment]
@matthiasblaesing I think the problem is not the same, I can go to the declarations from the .xhtml file.

My steps to reproduce the problem are similar to @SimonePadovan : 
 - I created a new project in NB with jdk 22, 
 - Created an empty .xhtml file, 
 - After this NB shows the exception I mentioned above.

**Comment by matthiasblaesing** [comment]
Still no reproducible for me. Please provide full message.log maybe there is something visible there.
See: https://stackoverflow.com/questions/50462124/netbeans-where-to-find-the-ide-log

**Comment by SimonePadovan** [comment]
Hi Matthias, i've attached my ide log, thank u very much
[log.txt](https://github.com/user-attachments/files/15757096/log.txt)

**Comment by dneves57** [comment]
@matthiasblaesing Here are the netbeans log messages.
[messages.log](https://github.com/user-attachments/files/15785182/messages.log)
The reported exception starts on line 906.

Thanks for your help

**Comment by matthiasblaesing** [comment]
@dneves57 @SimonePadovan please see if #7464 improves the situation. A nightly build is available from the "Checks" page or directly via: https://github.com/apache/netbeans/suites/24837488249/artifacts/1595526139

**Comment by SimonePadovan** [comment]
@matthiasblaesing  thank you, it seems to work fine with this nightly build.

**Comment by dneves57** [comment]
@SimonePadovan yes, it seems to have solved the problem. Thank you very much!

**Comment by zufolo441** [comment]
I confirm that this dev build works, solves this problem.

**Comment by kmrahe** [comment]
I just wanted to note that while the original report indicates this issue did not exist in NetBeans 21, I have indeed experienced it in that version on the same workstation in which I first experienced it in NetBeans 22. On two other workstations on which I use NB 21, however, I have not encountered it. The issue occurs in NB 22 on every workstation I've tried it on, though. I did not experiment with any versions older than 21.
